build.js

```diff
@@ -27,24 +30,59 @@ await build({
     {
       name: 'alias',
       setup({ onResolve, resolve }) {
-        onResolve({ filter: /^fs$/ }, () => ({
-          path: fileURLToPath(new URL('src/stubs/fs.cjs', import.meta.url)),
-          sideEffects: false,
-        }));
-        onResolve({ filter: /^util$/ }, () => ({
-          path: fileURLToPath(new URL('src/stubs/util.cjs', import.meta.url)),
-          sideEffects: false,
-        }));
-        onResolve({ filter: /^vscode-emmet-helper-bundled$/ }, () => ({
-          path: fileURLToPath(new URL('src/stubs/noop.cjs', import.meta.url)),
+        // These packages are imported, but can be stubbed.
+        onResolve(
+          {
+            filter:
+              /^(detect-indent|chalk|fs|path|util|util-deprecate|vscode-emmet-helper-bundled)$/,
+          },
+          ({ path }) => ({
+            path: fileURLToPath(new URL(`src/stubs/${path}.ts`, import.meta.url)),
+            sideEffects: false,
+          }),
+        );
+
+        // The tailwindcss main export exports CJS, but we can get better tree shaking if we import
+        // from the ESM src directoy instead.
+        onResolve({ filter: /^tailwindcss$/ }, ({ path, ...options }) =>
+          resolve('tailwindcss/src', options),
+        );
+        onResolve({ filter: /^tailwindcss($|\/lib)/ }, ({ path, ...options }) =>
+          resolve(path.replace('lib', 'src'), options),
+        );
+
+        // The tailwindcss-language-service main export exports CJS by default, but we get better
+        // tree shaking if we import the ESM variant.
+        onResolve({ filter: /^tailwindcss-language-service$/ }, ({ path, ...options }) =>
+          resolve('tailwindcss-language-service/dist/tailwindcss-language-service.esm.js', options),
+        );
+
+        // This file pulls in a number of dependencies, but we don’t really need it anyway.
+        onResolve({ filter: /^\.+\/(util\/)?log$/ }, ({ path, ...options }) =>
+          options.importer.includes(`${sep}tailwindcss${sep}`)
+            ? {
+                path: fileURLToPath(new URL('src/stubs/tailwindcss/utils/log.ts', import.meta.url)),
+                sideEffects: false,
+              }
+            : resolve(path, options),
+        );
+
+        // The culori main export exports CJS by default, but we get better tree shaking if we
+        // import the ESM variant.
+        onResolve({ filter: /^culori$/ }, ({ path, ...options }) =>
+          resolve('culori/build/culori.js', options),
+        );
+
+        // Some dependencies depend on very specific versions of postcss-selector-parser, but they
```

Comment:
wouldnt it be better to use the `resolve` field in package json for only even installing one version?

Reply:
I believe this is specific to yarn, but I’d love to be proven wrong

Reply:
hmm pnpm has `pnpm.overrides` or similar - leme check there must be something for npm ...

Reply:
its `overrides` https://github.com/npm/cli/releases/tag/v8.3.0
https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

v8.3.0

Reply:
Oh awesome!